data/DataHandler.php

```diff
@@ -0,0 +1,152 @@
+<?php
+//==================================
+// Class Definition of DataHandler
+//==================================
+
+require __DIR__.'/../utility/Equipment.php';
+require __DIR__.'/../utility/Room.php';
+
+define ('WAREHOUSE', 'warehouse');
+define("MAX_SPACE", 1000);
+
+class DataHandler {
+    
+    private array $room_list;
+    private array $equip_list;
+    
+    /**
+     * Constructor for the Handler.
+     * Initializes two arrays containing all equipment and all rooms.
+     * Automatically creates the warehouse.
+     */
+    public function __construct() {
+        $this->room_list = [WAREHOUSE => new Room(WAREHOUSE, MAX_SPACE)];
+        $this->equip_list = [];
+    }
+
```

Comment:
just fyi, not a must to implement
  public function __construct(private $equip_list = []) {
         $this->room_list = [WAREHOUSE => new Room(WAREHOUSE, MAX_SPACE)];
     }


Unless u want to force it to start empty.

There is a problem here -> u will find it later on, when u hook this to the storage itself (SESSION)